gradio/utils.py

```diff
@@ -646,11 +650,14 @@ def wrapper(*args, **kwargs):
         return wrapper
 
 
-def get_function_with_locals(fn: Callable, blocks: Blocks, event_id: str | None):
+def get_function_with_locals(
+    fn: Callable, blocks: Blocks, event_id: str | None, in_event_listener: bool
+):
     def before_fn(blocks, event_id):
         from gradio.context import thread_data
 
         thread_data.blocks = blocks
+        thread_data.in_event_listener = in_event_listener
```

Comment:
I'm trying to understand -- when is `thread_data.in_event_listener` set to `True`?